### PR TITLE
dev-games/ogre: 1.12.8 / 2.1 version bumps plus fixes/additions

### PR DESCRIPTION
**Pull request by Yamakuzure**
## Version 1.12.8 Changes:
 * imgui version has been raised to 1.76
 * gles2 USE flag has been removed. Ogre with GLES2/3 support now depends on HLSL2GLSL unconditionally. The flag and its config options have been 'backed up' as comments for easy re-integration once we have an ebuild for HLSL2GLSL. ( See: https://github.com/aras-p/hlsl2glslfork )
 * 'experimental' USE flag renamed to 'deprecated'. The HLMS component has switched from experimental to deprecated. Strange...
 * quakemap.cfg is gone
 * tests.cfg has been added

Fixes:
 * The 'Simple' demo has been fixed to compile with pedantic resource manager.

## Version 2.1 Changes from the pre-release:
 * gles2 USE flag is gone, the GLES2 render system is not ported, yet
 * Added "fine-granularity" USE flag to enable fine light mask granularity which has a noticable performance impact.
 * double-precision USE flag is back in. But if activated, both OGRE_SIMD_NEON and OGRE_SIMD_SSE2 must be disabled, so the performance impact is quite big.
 * mobile USE flag is gone, this needs further testing. Also I think that building for mobile devices is not really interesting on Gentoo, is it?
 * Fixed a few warnings and issues, and spoke with upstream about them. ( See: OGRECave/ogre-next#90 )
* Fixed a missing include compilation issue and reported it to upstream. ( See: OGRECave/ogre-next#132 )

##  metadata.xml: Specify what the "tools" USE flag does
    
Users should know, that the "tools" USE flag builds and install the
* MeshUpgrader,
* VRMLConverter and
* XMLConverter
tools. Thus users not in need of either can safely deactivate that USE flag.

Bug: https://bugs.gentoo.org/600734
Bug: https://bugs.gentoo.org/728458
Closes: https://bugs.gentoo.org/600734
Closes: https://bugs.gentoo.org/728458
Package-Manager: Portage-2.3.101, Repoman-2.3.22
Signed-off-by: Sven Eden <yamakuzure@gmx.net>

### CODE REVIEW

**Comment by Yamakuzure** [comment]
@chewi : Here is a nice little update. Alle Samples work fine on my Intel/Iris system.

This bump is a bit sour. To have the possibility to choose GLES2/3 over OpenGL, we now need HLSL2GLSL, so gles2 USE flag is off the table.
I'll try to add an ebuild for HLSL2GLSL soon, and would then bump out an -r1 of Ogre-1.12.6. (Unless they bring out 1.12.7 in the meantime...)

Hopefully I didn't overlook anything, and am looking forward to some peer review. :wink:

Thanks in advance!

**Comment by Yamakuzure** [comment]
Oh, dammit. I shouldn't have removed the opengl USE flag. I'll fix that...

**Comment by Yamakuzure** [comment]
Just a quick note: Ogre has its first major release for the new gpu-based 2.x series.
See here: https://github.com/OGRECave/ogre-next/releases/tag/2.1

I would like to bump this also, so please don't merge right now!

**Comment by Yamakuzure** [comment]
@chewi : I have finished the ogre-2.1.ebuid, and had a nice talk with @darksylinc who helped me greatly sorting out the numerous config options and some issue. (See https://github.com/OGRECave/ogre-next/issues/90)

My ebuilds are now ready for review. In the meantime I'll see what I can do about HLSL2GLSL.

**Comment by Yamakuzure** [comment]
Sorry, just replaced a hot-fix with a real fix.

I can now say, that the patches for both 1.12.6 and 2.1 are free off hot-fixes and only include real fixes that will be included in the next releases.

**Comment by Yamakuzure** [comment]
@chewi @SoapGentoo @steils @vapier 
Sorry for pinging you all, but I have no idea who's available and who's not thanks to Corona.

I would like to add the next level of the ogre engine, [Ogre-2.2.0 Cerberus](https://github.com/OGRECave/ogre-next/releases/tag/v2.2.0), but not before this PR is merged.

As always I assume that there is something that needs to be fixed first, but I daresay Chewi already helped me a great deal to eradicate most issues last time. (Doesn't mean I couldn't have added new ones. :roll_eyes: :wink:)

Again, sorry for the poking, I bet your spare time has been trampled down by the criis just like mine...

**Comment by Yamakuzure** [comment]
Sorry to nag, but I can not do anything else until this is merged.

**Comment by Yamakuzure** [comment]
In the meantime Ogre-1.12.7 got released...

**Comment by Yamakuzure** [comment]
Guys, this PR blocks my local gentoo tree. I can not work on anything else until this PR isn't resolved.
Is there any chance this'll happen in the near future?

Thanks in advance!

**Comment by steils** [comment]
Thanks for the PR, it is a big job you have done. And the PR is rather big, I guess that's why it takes so long for us to review it. Sorry for the delay.

Just for the future reference. This PR proposes two independent changes (upgrade 1.* and 2.* ebuilds), so it could be two different PRs. It is easier for reviewers to work with smaller changes at a time.

I have begun reviewing. 1.12.8, looks so far so good. But could you add the origins of the patches? Are you the author, or somebody else? What are their upstream status? A link to the upstream git commit or PR would be nice.

File: dev-games/ogre/ogre-2.1.ebuild [hunk 1]
```diff
@@ -115,8 +134,16 @@ src_configure() {
...
 
+	# GLES2 is not supported, yet
+	#	-DOGRE_BUILD_COMPONENT_HLMS_PBS=$(         usex mobile no yes)
+	#	-DOGRE_BUILD_COMPONENT_HLMS_PBS_MOBILE=$(  usex mobile)
+	#	-DOGRE_BUILD_COMPONENT_HLMS_UNLIT=$(       usex mobile no yes)
+	#	-DOGRE_BUILD_COMPONENT_HLMS_UNLIT_MOBILE=$(usex mobile)
+	#	-DOGRE_BUILD_RENDERSYSTEM_GLES2=$(usex gles2)
+
 	# The double-precision mode can not be enabled, yet.
 	#	-DOGRE_CONFIG_DOUBLE=$(usex double-precision)
```
Comment (steils) [inline dev-games/ogre/ogre-2.1.ebuild:148]: Ogre wiki says you also need to
```
Disable OGRE_SIMD_NEON
Disable OGRE_SIMD_SSE2
```
Have you tried disabled them? If not, isn't this the cause for the failure to build with double precision enabled?

**Comment by Yamakuzure** [comment]
> Thanks for the PR, it is a big job you have done. And the PR is rather big, I guess that's why it takes so long for us to review it. Sorry for the delay.

No problem, I know how difficult these times are for everyone. Me included. :shrug: 

> Just for the future reference. This PR proposes two independent changes (upgrade 1.* and 2.* ebuilds), so it could be two different PRs. It is easier for reviewers to work with smaller changes at a time.

I'll keep that in mind.

> I have begun reviewing. 1.12.8, looks so far so good. But could you add the origins of the patches? Are you the author, or somebody else? What are their upstream status? A link to the upstream git commit or PR would be nice.

Only ogre-1.12.8-upgrade_imgui.patch is new and by me to upgrade the imgui version from 1.73 (quite old) to 1.76 (more stable and more recent).

The other three 1.12.8 patches are renamed from what is already in the tree, and they are needed for Ogre to play nicely with a system-wide install. The default is local in the build directory. (Those patches already existed for 1.9.0)

Four of the 2.1 patches are already in the tree, the other for are one backport (ogre-2.1-d1c1116.patch) and three Gentoo specific fixes for compilation.
Discussion about that is here: https://github.com/OGRECave/ogre-next/issues/90
The discussion is also mentioned in both the commit message and the PR description, but I didn't exactly point out which patches resulted from it.

File: dev-games/ogre/ogre-2.1.ebuild (hunk 1 above)
Comment (Yamakuzure) [reply dev-games/ogre/ogre-2.1.ebuild:148]: I more hooked up on the very first sentence:
> We're not yet officially maintaining double precision, but it works (mostly?)

Well, double-precision already has a performance impact. Deactivating NEON/SSE2 for it to work doesn't sound wise to me.

However, I hit another problem. Running kernel-5.7+ means, that sysctl.h has been moved to /usr/inlcude/linux, and the build searches it in /usr/include/sys and fails. Dang.
This means I'll have to get a fix for kernels having it in there. AFAIR sys/sysctl.h is an Apple thing and irrelevant for Linux.

Give me a moment...

Comment (Yamakuzure) [reply dev-games/ogre/ogre-2.1.ebuild:148]: Reported here: https://github.com/OGRECave/ogre-next/issues/132

**Comment by chewi** [comment]
Merged! I'll try not to let it drag on so long next time.

In future, avoid strings like `Fixed:` in your commit messages because our git server tries to interpret these.

**Comment by Yamakuzure** [comment]
Thanks you very much, @chewi !
I'll keep it to "Bug", "Closes" and "Signed-off-by" in the future. :+1:

**Pull request closed without merging**